samza-rest/src/main/java/org/apache/samza/monitor/SamzaMonitorService.java

```diff
@@ -65,6 +66,15 @@ public void start() {
                     LOGGER.info("Scheduling monitor {} to run every {} ms", monitorName, schedulingIntervalInMs);
                     // MetricsRegistry has been added in the Monitor interface, since it's required in the eventual future to record metrics.
                     // We have plans to record metrics, hence adding this as a placeholder. We just aren't doing it yet.
+
+                    /**
+                     * Add a random jitter from the range [0, schedulingIntervalInMs] to the monitor
+                     * scheduling interval.
+                     */
+                    if (monitorConfig.isJitterEnabled()) {
```

Comment:
Not sure why you need SecureRandom, just use Random. Also, make it a private static final.

If nextInt returns an int, it can be negative. With 100% jitter range, you can end up scheduling tasks that run every 1 ms, even though they were meant to run every x minutes/hours/days. Clearly not what we want. Use a jitter.percent config instead.

Reply:
Correction: in the second part, nextInt it's b/w 0 and upper bound. So it can be up to 2x the original amount. Still better to use %.

Reply:
- Added a jitter percent, which controls the randomness that gets added to scheduling time.
- SecureRandom is a strong random number generator where as Random is pseudorandom number generator. It's better and proven to provide a more uniform distribution of a random numbers in a interval. 
- Made it as a static constant.